guides/unittesting-django-wagtail.md

```diff
@@ -0,0 +1,451 @@
+    
+    ```python
+    class TestMetaImage(TestCase):
+        def setUp(self):
+            self.preview_image = mommy.prepare(CFGOVImage)
+            self.social_sharing_image = mommy.prepare(CFGOVImage)
+
+        def test_meta_image_both(self):
+            page = mommy.prepare(
+                AbstractFilterPage,
+                social_sharing_image=self.social_sharing_image,
+                preview_image=self.preview_image
+            )
+            self.assertEqual(page.meta_image, page.social_sharing_image)    
+    ```
+
+- Creating specific instances of models in code. All other test data needs are generally accomplished by creating instances of models directly. 
+
+    For example, our [Django-Flags `DatabaseFlagsSoruce` test case](https://github.com/cfpb/django-flags/blob/master/flags/tests/test_sources.py#L46-L56) tests that a database-stored `FlagState` object is provided by the `DatabaseFlagsSource` class. To do that it creates a specific instance of `FlagState` using `FlagStage.objects.create()` that it expects to be returned:
+
+
+    ```python
+    class DatabaseFlagsSourceTestCase(TestCase):
+
+        def test_get_flags(self):
+            FlagState.objects.create(
+                name='MY_FLAG',
+                condition='boolean',
+                value='False'
+            )
+            source = DatabaseFlagsSource()
+            flags = source.get_flags()
+            self.assertEqual(flags, {'MY_FLAG': [Condition('boolean', 'False'), ]})
+    ```
+
+
+### Overriding settings
+
+It is sometimes useful to modify Django settings when testing code that may behave differently with different settings. To do this there are two decorators that can be applied to either an entire `TestCase` class or to individual test methods:
+
+- [`@override_settings()`](https://docs.djangoproject.com/en/1.11/topics/testing/tools/#django.test.override_settings) will override the contents of a setting variable.
+
+  We use `@override_settings` any time the outcome of the code being tested depends on a settings variable. 
+  
+  This can include testing different values of feature flags: 
+  
+  ```python
+  @override_settings(FLAGS={'MYFLAG': {'boolean': 'True'}})
+  ```
+  
+  Or when we need to test behavior with AWS S3:
+  
+  ```python
+  @override_settings(AWS_STORAGE_BUCKET_NAME='test.bucket')
+  ```
+  
+- [`@modify_settings()`](https://docs.djangoproject.com/en/1.11/topics/testing/tools/#django.test.modify_settings) will modify specific values within a list setting. 
+
+### Mocking
+
+We use the [Python Mock library](https://docs.python.org/3/library/unittest.mock.html) when we need to. We recommend mocking for:
+
+- External service calls, for example Akamai caching or GovDelivery (either their APIs or via `requests`). 
```

Comment:
I would prefer not to recommend using mocks for GovDelivery -- we have our own mock class for this which is already documented [here](https://cfpb.github.io/cfgov-refresh/testing-be/#govdelivery).

Regarding mocking `requests`, I quite like using [responses](https://github.com/getsentry/responses) instead of mock, which we do [in a few places already](https://github.com/cfpb/cfgov-refresh/blob/ae3f96a9482dd753ce430177c8243d82fc7319d9/cfgov/legacy/tests/views/test_complaint.py#L67). This makes it easy to test different response codes and content.

Reply:
Ah! I didn't realize we were already using responses!